protocol/messenger_messages.go

```diff
@@ -125,6 +125,18 @@ func (m *Messenger) DeleteMessageAndSend(ctx context.Context, messageID string)
 			}
 		}
 
+		if message.MessageType == protobuf.MessageType_PRIVATE_GROUP || message.MessageType == protobuf.MessageType_PUBLIC_GROUP {
+			group, err := newProtocolGroupFromChat(chat)
+			if err != nil {
+				return nil, err
+			}
+			admins := group.Admins()
+			canDeleteMessageForEveryone = stringSliceContains(admins, common.PubkeyToHex(&m.identity.PublicKey))
```

Comment:
@yqrashawn, given the body of the `if` is basically the same here as in `messenger_handler.go`, what do you think of creating a single function and parameterize the second argument to `stringSliceContains` (the only difference)?

It looks like a true duplication, so whenever someone changes A will necessarily need to change B, so that's why I'm suggesting this refactoring.

Reply:
Agreed. Same thing for the community block above, so the whole validation of `can delete` can be extracted.